autocards.py

```diff
@@ -1,63 +1,285 @@
+            print("No qa generated yet!")
+            return None
+
+        res = []
         for qa_pair in self.qa_pairs:
             if jeopardy:
-                print('\"' + prefix + qa_pair['answer'] + '\",\"' + qa_pair['question'] + '\"')
+                string = f"\"{prefix}{qa_pair['answer']}\",\"\
+{qa_pair['question']}\""
             else:
-                print('\"' + prefix + qa_pair['question'] + '\",\"' + qa_pair['answer'] + '\"')
+                string = f"\"{prefix}{qa_pair['question']}\",\"\
+{qa_pair['answer']}\""
+            res.append(string)
+        return res
 
-    def export(self, filename, prefix='', jeopardy=False):
-        if prefix != '':
+    def print(self, *args, **kwargs):
+        "Print qa pairs to the user"
+        print(self.string_output(*args, **kwargs))
+
+    def pprint(self, *args, **kwargs):
+        "Prettyprint qa pairs to the user"
+        pprint(self.string_output(*args, **kwargs))
+
+    def pandas_output(self, prefix=''):
+        if len(self.qa_pairs) == 0:
+            print("No qa generated yet!")
+            return None
+        "Output a Pandas DataFrame containing qa pairs and metadata"
+        df = pd.DataFrame(columns=list(self.qa_pairs[0].keys()))
+        for qa in self.qa_pairs:
+            df = df.append(qa, ignore_index=True)
+        for i in df.index:
+            for c in df.columns:
+                if pd.isna(df.loc[i, c]):
+                    # otherwise export functions break:
+                    df.loc[i, c] = "Error"
+        return df
+
+    def csv_export(self, filename, prefix='', jeopardy=False):
+        "Export qa pairs as csv file"
+        if len(self.qa_pairs) == 0:
+            print("No qa generated yet!")
+            return None
+        if prefix != "" and prefix[-1] != ' ':
             prefix += ' '
 
-        with open(filename, 'w', newline='') as file:
-            writer = csv.writer(file)
-            for qa_pair in self.qa_pairs:
-                if jeopardy:
-                    writer.writerow([prefix + qa_pair['answer'], qa_pair['question']])
-                else:
-                    writer.writerow([prefix + qa_pair['question'], qa_pair['answer']])
+        df = self.pandas_output(prefix)
+
+        def _remove_commas(string):
+            return string.replace(",", r"\,")
+
+        for i in df.index:
+            for c in df.columns:
+                df.loc[i, c] = _remove_commas(df.loc[i, c])
+
+        df.to_csv(filename)
+        print(f"Done writing qa pairs to {filename}")
+
+    def json_export(self, filename, prefix='', jeopardy=False):
+        "Export qa pairs as json file"
+        if len(self.qa_pairs) == 0:
+            print("No qa generated yet!")
+            return None
+        if prefix != "" and prefix[-1] != ' ':
+            prefix += ' '
+
+        self.pandas_output(prefix).to_json(filename)
+        print(f"Done writing qa pairs to {filename}")
+
+
+#def _debug_signal_handler(signal, frame):
```

Comment:
Maybe you meant to remove this?

Reply:
I will remove this commented section. Unless you were talking about the print?